src/components/views/rooms/EventTile.tsx

```diff
@@ -947,13 +947,26 @@ export class UnwrappedEventTile extends React.Component<IProps, IState> {
     };
 
     private showContextMenu(ev: React.MouseEvent, showPermalink?: boolean): void {
+        // Return if message right-click context menu isn't enabled
         if (!SettingsStore.getValue("feature_message_right_click_context_menu")) return;
+
+        // Return if we're in a browser and click either an a tag or we have
+        // selected text, as in those cases we want to use the native browser
+        // menu
+        const clickTarget = ev.target as HTMLElement;
+        if (
+            !PlatformPeg.get().allowOverridingNativeContextMenus() &&
```

Comment:
```suggestion
            !PlatformPeg.get().allowOverridingNativeContextMenus() ||
```

to maintain the check that was removed later on?

Reply:
We don't want to maintain that check - we want to allow the menu even in the browser; we just want to make sure we don't show the custom menu in the browser when the other conditions are met

Reply:
but surely if the user has highlighted text or right clicked a link then we'd want to show the OS menu on desktop too?

Reply:
[The agreement seemed to be to replace the OS menu on the desktop with the custom one](https://user-images.githubusercontent.com/60790/159077851-ee96736b-d568-49c1-8241-8192b034ef8e.png), I am not sure if there have been any developments, I assume this picture is from an internal repo, to which I don't have access...

Reply:
hmm, alright. The image is from an internal repo (where there's not much more information than the screenshot).